leakcanary/leakcanary-android-core/src/main/java/leakcanary/internal/activity/screen/HeapDumpsScreen.kt

```diff
@@ -111,5 +114,10 @@ internal class HeapDumpsScreen : Screen() {
         )
         countView.text = count
       }
+
+    if (listViewState != null) {
+      listView.onRestoreInstanceState(listViewState)
+    }
+    onScreenExiting { listViewState = listView.onSaveInstanceState() }
```

Comment:
Thanks, but this is a bit surprising: there's already a backstack mechanism built in to save and restore view state. When we create a new `BackstackFrame` instance, the `BackstackFrame` constructor invokes `view.saveHierarchyState()`. When we go back and bring a view back to be current, `NavigatingActivity.goBack()` `latest.restore(currentView)` which calls `view.restoreHierarchyState(viewState)`

Before adding this new behavior, I'd want to understand why the normal view state saving behavior isn't working here.

Reply:
I debugged through this, and the explanation here is that the view state is getting restored when we navigate back, but at that point the listview has no adapter. We load the data from the db async and restore the view state afterwards.